articles/quickstart/native/ios-objc/03-user-sessions.md

```diff
@@ -0,0 +1,282 @@
+:::
+
+## On Login: Store the user's token
+
+We will store the [accessToken](/tokens/access-token) **upon a successful login**, in order to prevent the user from being asked for login credentials again every time the app is re-launched.
+
+${snippet(meta.snippets.setup)}
+
+Then present the hosted login screen, like this:
+
+```objc
+HybridAuth *auth = [[HybridAuth alloc] init];
+[auth showLoginWithScope:@"openid profile" connection:nil callback:^(NSError * _Nullable error, A0Credentials * _Nullable credentials) {
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if (error) {
+            NSLog(@"Error: %@", error);
+        } else if (credentials) {
+          // Do something with credentials e.g.: save them.
+          // Auth0 will dismiss itself automatically by default.
+        }
+    });
+}];
+```
+
+You will want to store the `accessToken` value, which is inside the `Credentials` instance. To do so, you will use an `A0SimpleKeychain` instance:
+
+```objc
+A0SimpleKeychain *keychain = [[A0SimpleKeychain alloc] initWithService:@"Auth0"];
+[keychain setString:accessToken forKey:@"access_token"];
+```
+
+As you can see, `SimpleKeychain` can be seen simply as a key-value storage.
+
+## On Startup: Check accessToken existence
+
+The main purpose of storing this token is to save the user from having to re-enter login credentials upon relaunch of the app. So, **once the app has launched**, we need to check for the existence of an `accessToken` to see if we can automatically log the user in and redirect the user straight into the app's main flow, skipping any login screen.
+
+To do so, first, you retrieve its value from the `accessToken` key stored in the keychain:
+
+```objc
+A0SimpleKeychain *keychain = [[A0SimpleKeychain alloc] initWithService:@"Auth0"];
+NSString* accessToken = [keychain stringForKey:@"access_token"];
+if (accessToken) {
+    // accessToken exists
+    // You still need to validate it!
+}
+```
+
+## Validate an accessToken
+
+Then, if such a token exists, you need to check whether it's still valid, has expired, or is no longer valid for some other reason, such as being revoked. To do so, you will use `Auth0` to fetch the user's profile based on the `accessToken` we've got:
+
+```objc
+[auth userInfoWithAccessToken:accessToken callback:^(NSError * _Nullable error, A0Profile * _Nullable profile) {
+        if (error) {
+            // accessToken has expired or no longer valid
+        } else {
+            // The accessToken is still valid and you have the user's profile
+            // This would be a good time to store the profile
+        }
+}];
+```
+
+## Dealing with a non-valid accessToken
+
+How to deal with a non-valid accessToken is up to you. You will normally choose between two scenarios:
+Either you ask users to re-enter theirs credentials, or you can use `.renew(withRefreshToken: refreshToken)` with a [refresh_token((/refresh-token)) to obtain a new valid accessToken again.
```

Comment:
> re-enter theirs credentials 

should be `re-enter their credentials`

Also, the refresh token link formatting is off

>with a [refresh_token((/refresh-token))

`with a [refresh_token](/refresh-token)`